dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/DefaultIconService.java

```diff
@@ -206,27 +200,35 @@ private void validateIconKeyNotNullOrEmpty(String key) throws BadRequestExceptio
     if (Strings.isNullOrEmpty(key)) {
       throw new BadRequestException("Icon key not specified.");
     }
+
+    validateIconKey(key);
   }
 
   private FileResource getFileResource(String fileResourceUid)
       throws BadRequestException, NotFoundException {
     if (Strings.isNullOrEmpty(fileResourceUid)) {
-      throw new BadRequestException("File resource id not specified.");
+      throw new BadRequestException("FileResource id not specified.");
     }
 
     Optional<FileResource> fileResource =
         fileResourceService.getFileResource(fileResourceUid, CUSTOM_ICON);
     if (fileResource.isEmpty()) {
-      throw new NotFoundException(
-          String.format("File resource %s does not exist", fileResourceUid));
+      throw new NotFoundException(String.format("FileResource %s does not exist", fileResourceUid));
     }
 
     return fileResource.get();
   }
 
-  private CustomIcon validateCustomIconExists(String key)
-      throws NotFoundException, BadRequestException {
+  private Icon validateIconExists(String key) throws NotFoundException, BadRequestException {
     validateIconKeyNotNullOrEmpty(key);
-    return getCustomIcon(key);
+    return getIcon(key);
+  }
+
+  private Icon validateIconExists(Icon icon) throws NotFoundException, BadRequestException {
+    if (icon == null) {
```

Comment:
There are a bunch of these methods that look like we are repeating ourselves. Can this not be `!validateIconExists(icon)`?